esphome/components/as3935/as3935.cpp

```diff
@@ -216,6 +223,104 @@ uint32_t AS3935Component::get_lightning_energy_() {
   return pure_light;
 }
 
+// REG0x03, bit [7:6], manufacturer default: 0 (16 division ratio).
+// This function returns the current division ratio of the resonance frequency.
+// The antenna resonance frequency should be within 3.5 percent of 500kHz, and
+// so when modifying the resonance frequency with the internal capacitors
+// (tuneCap()) it's important to keep in mind that the displayed frequency on
+// the IRQ pin is divided by this number.
+uint8_t AS3935Component::get_div_ratio() {
+  ESP_LOGV(TAG, "Calling get_div_ratio");
+  uint8_t reg_val = this->read_register(INT_MASK_ANT);
+  reg_val &= ~DIV_MASK;
```

Comment:
There's the `read_register_` method, so that you don't have to do this masking manually everywhere:

```suggestion
  uint8_t reg_val = this->read_register_(INT_MASK_ANT, DIV_MASK);
```